File: java/leetcode/src/main/java/com/leetcode/BST/Search_in_a_Binary_Search_Tree_700.java
package com.leetcode.BST;

public class Search_in_a_Binary_Search_Tree_700 {
    public TreeNode searchBST(TreeNode root, int val) {
        if (root == null) return null;
        if (root.val == val) return root;
        if (val < root.val) {
            return searchBST(root.left, val);
        }
        return searchBST(root.right, val);
    }
}

File: js/algrithm/BST/Search_in_a_Binary_Search_Tree_700.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var searchBST = function(root, val) {

    const dfs = function(node, val) {
        if (!node) return null
        if (node.val === val) {
            return node
        } else if (node.val > val) {
            return dfs(node.left, val)
        } else {
            return dfs(node.right, val)
        }
    }

    const ret = dfs(root, val)
    return ret
};